jekyll-run.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -euo pipefail
+
+docker run -it --rm -v ./static:/srv/jekyll jekyll /bin/bash -c "bundle exec jekyll build -w"
```

Comment:
The implicit order dependency between these scripts is something that I've been bitten by enough times I like to try to at least make it explicit: instead of assuming the `jekyll` image exists & is up-to-date, I'd suggest something like:

```suggestion
docker run -it --rm -v ./static:/srv/jekyll "$1" /bin/bash -c "bundle exec jekyll build -w"
```

Which, in combination with the earlier suggestion, lets you encode this into your shell history:

```
./jekyll-run.sh $(./jekyll-build.sh --quiet)
```

which expresses the dependency in a way future-you will probably thank you for. If you need offline development, you can still do:

```
./jekyll-build.sh -t jekyll-builder-$(date +'%Y-%m-%d')
```

ahead of time, and then e.g.

```
./jekyll-run.sh jekyll-builder-2024-04-06
```

(or even `./jekyll-run.sh <sha....>`)

to use the local image when you're offline.

Reply:
oh does the build step (docker build) output the tag or sha or whatever to the shell?

Reply:
yeah, it does, though it's a real PITA to parse it without using `--quiet`, which you generally don't want because it gives no feedback. From `man docker-build`:

```
       -q, --quiet true|false
          Suppress the build output and print image ID on success. The default is false.
```

So what I usually do is actually something like `build && run $(build -q)`—the first gives me feedback, and the second poops out the image ID (sha) for real cheap because of all the caching buildkit does. There's other ways, but that's the one my fingers remember first.